src/main/java/com/example/springboot/SpotifyAuthController.java

```diff
@@ -0,0 +1,50 @@
+package com.example.springboot;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+import com.example.springboot.SpotifyConfig;
+
+import java.util.Map;
+
+@RestController
+public class SpotifyAuthController {
+  @Autowired
+  private SpotifyConfig spotifyConfig;
+
+  @PostMapping("/api/auth/spotify")
+  public ResponseEntity<?> authenticateSpotify(@RequestBody Map<String, String> payload) {
+    String code = payload.get("authCode");
+
+    RestTemplate restTemplate = new RestTemplate(); 
+
+    HttpHeaders headers = new HttpHeaders(); 
+    headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+
+    MultiValueMap<String, String> requestBody = new LinkedMultiValueMap<>();
+
+    requestBody.add("grant_type", "authorization_code");
+    requestBody.add("code", code);
+    requestBody.add("redirect_uri", spotifyConfig.getRedirectUri());
+    requestBody.add("client_id", spotifyConfig.getClientId());
+    requestBody.add("client_secret", spotifyConfig.getClientSecret());
+
+    HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(requestBody, headers);
+
+    // TODO: Getting errors here from the Spotify endpoint for invalid auth code
+    ResponseEntity<Map> response = restTemplate.postForEntity(
```

Comment:
Update, this is not an error. React has this silly thing called "strict mode" where they rerender the page twice only in development mode to help devs find issues quicker. So this endpoint was being hit twice and the second time would get rejected and thus throw an error. This is not a BE issue